### ISSUE TITLE
Did it working with last swagger 3.0

### ISSUE THREAD

**Issue by simonjoom** [feathersjs/feathers-swagger#61]
this thread is not a bug but i want to share my work to make it working with version 3.0
installed swagger-ui v2 (then overrided the folder dist of the last v3 

Change to do in feathers-swagger/lib/index.js:

var rootDoc = Object.assign({ 
...
      swagger: '3.0',
...
      ignore: {
        tags: []
      }
    }, config || {});

in each parameters change
{...
            name: 'resourceId',
            type: 'integer'
...}
to
{...
            name: '_id',
            type: 'string'
....}

then
then i overrided node_modules/feathers-swagger/dist/ from the last dist v3 of swagger
change node_modules/feathers-swagger/dist/

in configure:
foce uiIndex to be in dist of your modules
var modulesRootPath = require.resolve('swagger-ui');
modulesRootPath = modulesRootPath.substr(0, modulesRootPath.lastIndexOf('node_modules'));
.configure(swagger({
    docsPath: '/docs',
    uiIndex: path.join(modulesRootPath, 'node_modules/swagger-ui/dist/index.html'),
}))

I tested it's working well with the mods (i tested in https ) .

**Comment by christopherjbaker** [comment]
@simonjoom I am looking into upgrading, but am having a small issue. I was wondering if you could offer some insight. The html file that is in the dist folder of swagger 2 is not present in 3. You mentin "foce uiIndex to be in dist of your modules", and your code references `node_modules/swagger-ui/dist/index.html`, which does not exist for me in swagger-ui 3.

**Comment by simonjoom** [comment]
Me i started from a swagger-ui 2  to pass to 3 .
 so i didn't install swagger-ui v3 with npm . Maybe swagger-ui v3 don't install dist with npm 

Just go in https://github.com/swagger-api/swagger-ui (it's the last swagger)
download it and copy the dist folder in the location /node_module where you installed your swagger-ui

In fact you have no need other code than the code inside this dist to launch swagger-ui v3 (and the description i did upper to run it).

**Comment by simonjoom** [comment]
just a comment, i use feathers-swagger to help debugging my api it's nice, 
I use as well swagger-mongoose (that i did some mods) to create my db schemas completly in swagger format, and to transcript them to mongoose, 
the 2 tools togethers are nice to create easy mongoose dtb relation

**Comment by kristianmandrup** [comment]
@simonjoom Could you share this with us. Thanks :)

**Comment by kristianmandrup** [comment]
When upgrading to Swagger UI 3.x, use [swagger-ui-dist](https://www.npmjs.com/package/swagger-ui-dist)

In `package.json`

```json
    "swagger-ui": "^3.2.0",
    "swagger-ui-dist": "^3.2.0"
```

In `src/index.js` reference `swagger-ui-dist`

```js
// Find node_modules root path
let modulesRootPath = require.resolve('swagger-ui-dist');
modulesRootPath = modulesRootPath.substr(0, modulesRootPath.lastIndexOf('node_modules'));

export {
  modulesRootPath
}

export default function init(config) {
  return function () {
    // ...
    const rootDoc = Object.assign({
      paths: {},
      definitions: {},
      swagger: '3.2',  // <-- upgrade version
   // ...
```

In `app.js` of your feathers app change the `uiIndex` config option to reference `swagger-ui-dist/index.html`

```js
const swagger = require('../../lib');

  .configure(swagger({
    docsPath: '/docs',
    uiIndex: path.join(swagger.modulesRootPath, 'node_modules/swagger-ui-dist/index.html'),
```

**Comment by kristianmandrup** [comment]
To customize the `index.html` file referenced, edit this as you like. Specifically set the `url` to point to your own feathers API and not the petstore

```js
      const ui = SwaggerUIBundle({
        url: "http://petstore.swagger.io/v2/swagger.json",
```

```html
<!-- HTML for static distribution bundle build -->
<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Swagger UI - Simple</title>
  <link href="https://fonts.googleapis.com/css?family=Open+Sans:400,700|Source+Code+Pro:300,600|Titillium+Web:400,600,700"
    rel="stylesheet">
  <link rel="stylesheet" type="text/css" href="./swagger-ui.css">
  <link rel="icon" type="image/png" href="./favicon-32x32.png" sizes="32x32" />
  <link rel="icon" type="image/png" href="./favicon-16x16.png" sizes="16x16" />
  <style>
    html {
      box-sizing: border-box;
      overflow: -moz-scrollbars-vertical;
      overflow-y: scroll;
    }

    *,
    *:before,
    *:after {
      box-sizing: inherit;
    }

    body {
      margin: 0;
      background: #fafafa;
    }

  </style>
</head>

<body>

  <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" style="position:absolute;width:0;height:0">
    <defs>
      <symbol viewBox="0 0 20 20" id="unlocked">
        <path d="M15.8 8H14V5.6C14 2.703 12.665 1 10 1 7.334 1 6 2.703 6 5.6V6h2v-.801C8 3.754 8.797 3 10 3c1.203 0 2 .754 2 2.199V8H4c-.553 0-1 .646-1 1.199V17c0 .549.428 1.139.951 1.307l1.197.387C5.672 18.861 6.55 19 7.1 19h5.8c.549 0 1.428-.139 1.951-.307l1.196-.387c.524-.167.953-.757.953-1.306V9.199C17 8.646 16.352 8 15.8 8z"></path>
      </symbol>

      <symbol viewBox="0 0 20 20" id="locked">
        <path d="M15.8 8H14V5.6C14 2.703 12.665 1 10 1 7.334 1 6 2.703 6 5.6V8H4c-.553 0-1 .646-1 1.199V17c0 .549.428 1.139.951 1.307l1.197.387C5.672 18.861 6.55 19 7.1 19h5.8c.549 0 1.428-.139 1.951-.307l1.196-.387c.524-.167.953-.757.953-1.306V9.199C17 8.646 16.352 8 15.8 8zM12 8H8V5.199C8 3.754 8.797 3 10 3c1.203 0 2 .754 2 2.199V8z"
        />
      </symbol>

      <symbol viewBox="0 0 20 20" id="close">
        <path d="M14.348 14.849c-.469.469-1.229.469-1.697 0L10 11.819l-2.651 3.029c-.469.469-1.229.469-1.697 0-.469-.469-.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-.469-.469-.469-1.228 0-1.697.469-.469 1.228-.469 1.697 0L10 8.183l2.651-3.031c.469-.469 1.228-.469 1.697 0 .469.469.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c.469.469.469 1.229 0 1.698z"
        />
      </symbol>

      <symbol viewBox="0 0 20 20" id="large-arrow">
        <path d="M13.25 10L6.109 2.58c-.268-.27-.268-.707 0-.979.268-.27.701-.27.969 0l7.83 7.908c.268.271.268.709 0 .979l-7.83 7.908c-.268.271-.701.27-.969 0-.268-.269-.268-.707 0-.979L13.25 10z"
        />
      </symbol>

      <symbol viewBox="0 0 20 20" id="large-arrow-down">
        <path d="M17.418 6.109c.272-.268.709-.268.979 0s.271.701 0 .969l-7.908 7.83c-.27.268-.707.268-.979 0l-7.908-7.83c-.27-.268-.27-.701 0-.969.271-.268.709-.268.979 0L10 13.25l7.418-7.141z"
        />
      </symbol>


      <symbol viewBox="0 0 24 24" id="jump-to">
        <path d="M19 7v4H5.83l3.58-3.59L8 6l-6 6 6 6 1.41-1.41L5.83 13H21V7z" />
      </symbol>

      <symbol viewBox="0 0 24 24" id="expand">
        <path d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z" />
      </symbol>

    </defs>
  </svg>

  <div id="swagger-ui"></div>

  <script src="./swagger-ui-bundle.js">


  </script>
  <script src="./swagger-ui-standalone-preset.js">


  </script>
  <script>
    window.onload = function () {

      // Build a system
      const ui = SwaggerUIBundle({
        url: "http://petstore.swagger.io/v2/swagger.json",
        dom_id: '#swagger-ui',
        deepLinking: true,
        presets: [
          SwaggerUIBundle.presets.apis,
          SwaggerUIStandalonePreset
        ],
        plugins: [
          SwaggerUIBundle.plugins.DownloadUrl
        ],
        layout: "StandaloneLayout"
      })

      window.ui = ui
    }

  </script>
</body>

</html>
```

**Comment by christopherjbaker** [comment]
@kristianmandrup I had started a fork with the upgrade with plans to PR, but some things came up. Are you planning to PR this? If not, may I use your above to code?

**Comment by kristianmandrup** [comment]
I have a working fork v3.2, posted it as the latest issue. Didn't do PR yet. Please try it out ;)
Also updated Readme with usage instructions.

**Comment by simonjoom** [comment]
Sorry kristian i have no time to try that, but i guess it should to work 👍  
About this work " use as well swagger-mongoose (that i did some mods) to create my db schemas completly in swagger format, and to transcript them to mongoose," 

I did some patch at swagger-mongoose (old repo) to complet all needs, i can share but cannot explain (it's old work)

below the code of the part interesting; 

> 'use strict';
> import swaggerMongoose from 'swagger-mongoose';
> import omit from 'lodash/omit';
> var fs = require('fs');
> var path = require('path');
> 
> const docs = {
>   swagger: 3,
>   definitions: {},
> }
> 
> const basename = path.basename(__filename);
> export default function init() {
>   const app = this;
>   let props = {
>   };
>   
>   if (app.db)
>     return db;
>   
>   take all schema from a directory and add them in the object 'props'
>   fs.readdirSync(__dirname)
>   .filter(function(file) {
>     return (file.indexOf('.') !== 0) && (file !== basename) && (file.slice(-3) === '.js');
>   }).forEach(function(file) {
>     const schemaProperty = require(`./${file}`).default;
>     props = Object.assign({}, props, schemaProperty);
>   });
>   
>   docs.definitions = props;
>   
>   // console.log( docs.definitions)
>   const db = swaggerMongoose.compile(docs, false, {});
>   db.docs = docs;
>   console.log("done dtb model")

>   app.db = db;
>   // return db;
> }
 
this code take a swagger definition to transpire in mongoose schema, so that your application you have no need to create the swagger definition and the mongo model.

This create the mongo model for your application feather to be used.
The relations between model works (i tested many time) and i can use the native function as $populate for my needs

To use swagger-mongoose you need just to add an extra definition in your model swagger:
like bellow:
    "x-swagger-mongoose": {
      "additional-properties": {
        "avatar": {
          "x-swagger-mongoose": {
            "$ref": "#/definitions/assets"
          }
        },
        "instructors": {
          "x-swagger-mongoose": {
            "$ref": "#/definitions/newinstructors"
          },
        },
        "clients": {
          "x-swagger-mongoose": {
            "$ref": "#/definitions/newclients"
          },
        }
      },
      "schema-options": {
        "timestamps": "true"
      }
    }

this keep the relations of the final mongoose dtb model created (so it's vey nice)

**Comment by daffl** [comment]
Closed via #79
